### PR TITLE
incorporate conformance/feature requirements

### PR DESCRIPTION
**Pull request by lionelvillard**

### CODE REVIEW

File: pkg/framework/test.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package framework
+
+// Test defines functions for configuring and running a single test case
+type Test interface {
+
+
```
Comment (mattmoor) [inline pkg/framework/test.go:22]: Format Go code:
```suggestion

```

**Review by mattmoor — COMMENTED** [review]
Produced via:
  `gofmt -s -w $(find -path './vendor' -prune -o -path './third_party' -prune -o -name '*.pb.go' -prune -o -type f -name '*.go' -print)`
  `goimports -w $(find -name '*.go' | grep -v vendor | grep -v third_party | grep -v .pb.go | grep -v wire_gen.go)`

**Comment by lionelvillard** [comment]
@dprotaso @n3wscott @devguyio

**Comment by n3wscott** [comment]
Sweeeet!!! I will look first thing tomorrow

File: pkg/framework/config.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ * limitations under the License.
+ */
+
+package framework
+
+// Config is the test suite configuration
+type Config interface {
+}
+
+
```
Comment (mattmoor) [inline pkg/framework/config.go:23]: Format Go code:
```suggestion

```

**Review by mattmoor — COMMENTED** [review]
Produced via:
  `gofmt -s -w $(find -path './vendor' -prune -o -path './third_party' -prune -o -name '*.pb.go' -prune -o -type f -name '*.go' -print)`
  `goimports -w $(find -name '*.go' | grep -v vendor | grep -v third_party | grep -v .pb.go | grep -v wire_gen.go)`

File: pkg/framework/config.go (hunk 2 above)
Comment (mattmoor) [inline pkg/framework/config.go:23]: Format Go code:
```suggestion

```

**Review by mattmoor — COMMENTED** [review]
Produced via:
  `gofmt -s -w $(find -path './vendor' -prune -o -path './third_party' -prune -o -name '*.pb.go' -prune -o -type f -name '*.go' -print)`
  `goimports -w $(find -name '*.go' | grep -v vendor | grep -v third_party | grep -v .pb.go | grep -v wire_gen.go)`

**Comment by n3wscott** [comment]
happy to start with this and we will make updates as we learn

this repo is PoC for testing until we decide and agree ... etc, exciting to get progress!!

/lgtm
/approve

**Pull request merged**